### PR TITLE
Dev

### PR DESCRIPTION
**Pull request by silverdaz**
- [x] Bug fix
- [x] Functional change
- [x] New feature
- [x] Code cleanup
- [x] Build system change
- [x] Documentation change
- [ ] Language translation

### Pull request long description:
* No passwords in environment variables.
* simplified configuration of each container
* network separation for docker-compose setting
* Index mirroring agent and queues
* secrets (password or private key) injected in only relevant containers
* Better logging
* More robust connections to DB and MQ
* Testing outgestion against CentralEGA file permissions
* Fundations for management routine for header reencryption.
* Support the updated version of Crypt4GH

### CODE REVIEW

**Comment by blankdots** [comment]
initial round of review comments also some `flake8 .` fixes available in https://github.com/NBISweden/LocalEGA/pull/374

File: deploy/images/README.md [hunk 1]
```diff
@@ -8,7 +8,7 @@ In the current folder, type `make` and the images are created in order.
 
 It takes some time.
 
-Later on, if the `crg/ega-*` do need to be recreated, one can type `make`.
```
Comment (jbygdell) [inline deploy/images/README.md]: This commit contains other changes that does not match the commit message,
these should be split out to a separate commit to minimise confusion.
